src/System.Text.Encoding.CodePages/src/System.Text.Encoding.CodePages.csproj

```diff
@@ -58,7 +57,15 @@
     </EmbeddedResource>
   </ItemGroup>
   <ItemGroup>
+    <None Include="Data\CodePageNameMappings.csv" />
+    <None Include="Data\PreferredCodePageNames.csv" />
     <None Include="project.json" />
   </ItemGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
+  <!-- Generator for code mapping table -->
+  <PropertyGroup>
+    <IANAMappingPath>$(MSBuildThisFileDirectory)\Data\CodePageNameMappings.csv</IANAMappingPath>
+    <PreferredIANANamesPath>$(MSBuildThisFileDirectory)\Data\PreferredCodePageNames.csv</PreferredIANANamesPath>
+  </PropertyGroup>
```

Comment:
As mentioned in your build tools PR I would add a custom target here that someone can call on demand whenever they want to update the file and it would update the file in the source tree and be committed.